### PR TITLE
Kraken: consider reduced_service as modifying effect

### PR DESCRIPTION
**Pull request by xlqian**
consider REDUCED_SERVICE as modifying effect 
This should fix the test in artemis: https://github.com/CanalTP/artemis/blob/master/artemis/tests/guichet_unique_test.py#L258

### CODE REVIEW

**Comment by xlqian** [comment]
I'm checking if this change may impact the disruptions from Chaos

File: source/kraken/realtime.cpp [hunk 1]
```diff
@@ -291,6 +291,7 @@ static const std::string get_wordings(nt::disruption::Effect effect) {
     case Effect::MODIFIED_SERVICE:
         return  "trip modified";
     case Effect::REDUCED_SERVICE:
+        return "reduced service";
     case Effect::ADDITIONAL_SERVICE:
```
Comment (pbougue) [inline source/kraken/realtime.cpp:296]: Could be nice to improve all wordings, for next debug need :)

File: source/kraken/apply_disruption.cpp [hunk 2]
```diff
@@ -501,12 +498,15 @@ static bool is_modifying_effect(nt::disruption::Effect e) {
     return in(e, {nt::disruption::Effect::NO_SERVICE,
                   nt::disruption::Effect::SIGNIFICANT_DELAYS,
                   nt::disruption::Effect::MODIFIED_SERVICE,
-                  nt::disruption::Effect::DETOUR});
+                  nt::disruption::Effect::DETOUR,
```
Comment (pbougue) [inline source/kraken/apply_disruption.cpp:501]: Not sure: Shouldn't we define it negatively?
We might end up with the same bug next time we add/handle a value of that enum, no?

**Review by pbougue — COMMENTED** [review]
This looks good.
Do we consider testing in artemis is enough or is it easy (and worthy) to add unit tests also?

**Comment by kadhikari** [comment]
A test with effect = REDUCED_SERVICE in integration test (kirin_realtile_tests.py) should work.

File: source/kraken/apply_disruption.cpp (hunk 2 above)
Comment (xlqian) [reply source/kraken/apply_disruption.cpp:501]: And I have another question at a higher level: **shouldn't this be handled by the connector?** this code is common for chaos and kirin, i've no idea what to do if we get a new connector and the definition of `modifying_effect` is totally defined in a new way...

**Review by TeXitoi — APPROVED** [review]

**Review by kadhikari — APPROVED** [review]

**Review by nonifier — APPROVED** [review]

**Comment by nonifier** [comment]
And here is the Artemis change that goes along: https://github.com/CanalTP/artemis/pull/280 :)

ps: I kicked off a [custom build on Artemis](https://ci.navitia.io/view/artemis/job/custom_artemis_reference_branch_and_artemis_branch_build/) (knowingly failing on [6 guichet unique tests](https://ci.navitia.io/view/artemis/job/Artemis/2000/#showFailuresLink)) 

ps 2: [Tests have failed](https://ci.navitia.io/view/artemis/job/custom_artemis_reference_branch_and_artemis_branch_build/182/#showFailuresLink)... Maybe due to the already failing tests on guichet-unique.

File: source/jormungandr/tests/kirin_realtime_tests.py [hunk 3]
```diff
@@ -1189,7 +1235,7 @@ def test_add_modify_and_delete_one_stop_time(self):
             == 'unchanged'  # Why ?
         )
         assert disrupts['disruptions'][11]['severity']['effect'] == 'DETOUR'
-        assert disrupts['disruptions'][11]['severity']['name'] == 'trip modified'
+        assert disrupts['disruptions'][11]['severity']['name'] == 'detour'
```
Comment (nonifier) [inline source/jormungandr/tests/kirin_realtime_tests.py:1238]: Consistency for the win ! :+1:

**Review by nonifier — APPROVED** [review]

**Review by kadhikari — APPROVED** [review]

File: source/kraken/realtime.cpp (hunk 1 above)
Comment (xlqian) [reply source/kraken/realtime.cpp:296]: done

**Pull request merged**